mne/channels/channels.py

```diff
@@ -722,7 +722,9 @@ def pick_types(self, meg=None, eeg=False, stim=False, eog=False,
             ias=ias, syst=syst, seeg=seeg, dipole=dipole, gof=gof, bio=bio,
             ecog=ecog, fnirs=fnirs, include=include, exclude=exclude,
             selection=selection)
-        return self._pick_drop_channels(idx)
+        self._pick_drop_channels(idx)
```

Comment:
More DRY to just call `self._pick_projs()` at the end of `self._pick_drop_channels`, no? The one you don't technically need it for is `reorder_channels`, but that's okay because it will be a no-op there. It would however have avoided the bug that seems to be here, namely, that it seems to be missing from `def drop_channels` (right?)

Reply:
> It would however have avoided the bug that seems to be here, namely, that it seems to be missing from `def drop_channels` (right?)

I had forgotten about `drop_channels`, yes, that's a bug.

Good idea you're proposing here, will do that!

Reply:
@larsoner I now included projector handling in `drop_channels` too.

After thinking about your more DRY proposal for a bit, I felt that I would like to keep `_pick_drop_channels` (which acts on channels) and `_pick_projs` (which acts on projectors) as two clearly separated steps for now, meaning I'd prefer keep the current non-DRY approach. It's just my personal feeling, I prefer not to convolute stuff too much. 

If you have strong objections, I can change the according like you suggested; otherwise, I'd like to keep it the way it currently is.